src/torrent.cpp

```diff
@@ -856,9 +927,12 @@ void Torrent::run() {
     for (auto& p : m_peers) {
       ran += p->io_service().poll_one();
     }
-    retry_pieces();
-    retry_peers();
-    // If no handlers ran, then sleep.
+    // TODO: Investigate why these were needed originally. They can
+    //       not be called like commented out here at least since it
+    //       will spam calls in every loop iteration of the run loop.
+    // retry_pieces();
+    // retry_peers();
```

Comment:
We might have to start this up outside the loop - or is it enough with the existing calls?

Reply:
Note that the rate limiting was removed, so we need to ensure we are not scheduling more calls than needed.